include/hipSYCL/sycl/libkernel/generic/hiplike/builtins.hpp

```diff
@@ -457,6 +457,11 @@ HIPSYCL_HIPLIKE_BUILTIN T __hipsycl_mul24(T x, T y) noexcept {
   return __mul24(x, y);
 }
 
+template<class T>
+HIPSYCL_HIPLIKE_BUILTIN T __hipsycl_popcount(T x) noexcept {
+  return __popc(x);
+}
```

Comment:
[`__popc` takes uint32](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__INT.html#group__CUDA__MATH__INTRINSIC__INT_1g43c9c7d2b9ebf202ff1ef5769989be46), so we need to specialize for 8-byte types.

```suggestion
template<class T, std::enable_if_t<std::is_integral_v<T> && sizeof(T) <= 4, int> = 0>
HIPSYCL_HIPLIKE_BUILTIN T __hipsycl_popcount(T x) noexcept {
  return __popc(static_cast<__hipsycl_uint32>(x));
}

template<class T, std::enable_if_t<std::is_integral_v<T> && sizeof(T) == 8, int> = 0>
HIPSYCL_HIPLIKE_BUILTIN T __hipsycl_popcount(T x) noexcept {
  return __popcll(static_cast<__hipsycl_uint64>(x));
}
```